torch/_refs/__init__.py

```diff
@@ -4025,6 +4026,29 @@ def arange(
     )
 
 
+@register_decomposition(torch.ops.aten.lerp)
+@out_wrapper()
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("start", "end", "weight"),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
+)
+def lerp(start: Tensor, end: Tensor, weight: Union[Tensor, NumberType]):
+    check(start.dtype == end.dtype, lambda: f"expected dtype {start.dtype} for `end` but got dtype {end.dtype}")
```

Comment:
too late for that after type promotion wrapper?

Reply:
I see this as "the reference implements an extended behaviour over the PyTorch op", as we support a few more combinations of dtypes.

Reply:
Ok, but this check in reference is then pointless, the inputs it's getting are all the same type.

Reply:
ah, I see what you mean. I'll clean this up later in this stack.

Reply:
Added a clean-up in https://github.com/pytorch/pytorch/pull/87183